integrations/access/accessrequest/app.go

```diff
@@ -352,16 +352,28 @@ func (a *App) getMessageRecipients(ctx context.Context, req types.AccessRequest)
 		recipientSet.Add(common.Recipient{})
 		return recipientSet.ToSlice()
 	case types.PluginTypeOpsgenie:
-		if recipients, ok := req.GetSystemAnnotations()[types.TeleportNamespace+types.ReqAnnotationSchedulesLabel]; ok {
-			for _, recipient := range recipients {
-				rec, err := a.bot.FetchRecipient(ctx, recipient)
-				if err != nil {
-					log.Warning(err)
-				}
-				recipientSet.Add(*rec)
+		// When both notify-services and approve-schedules are present, each is used for their own intended purpose.
+		recipients := make([]string, 0)
+		if approveSchedules, ok := req.GetSystemAnnotations()[types.TeleportNamespace+types.ReqAnnotationApproveSchedulesLabel]; ok {
+			if notifySchedules, ok := req.GetSystemAnnotations()[types.TeleportNamespace+types.ReqAnnotationNotifySchedulesLabel]; ok {
+				recipients = notifySchedules
+			} else {
+				// When only approve-schedules annotation is present
+				// it is used for both notifications and auto-approval to keep the behavior there is now.
+				recipients = approveSchedules
```

Comment:
This block does not cover the case where only `notify-services` is set. This function will return an empty recipient set while the service from the annotation should be notified.

According to the PR description, the correct behaviour is:

> just notify-schedules set - alert is created but no auto approval flow

This function should have tests. If you want to gain time here are the ones I wrote to review the PR:

<Details>

<summary>Git patch of the tests that you can "git apply"</summary>

```diff
diff --git a/integrations/access/accessrequest/app_test.go b/integrations/access/accessrequest/app_test.go
new file mode 100644
index 0000000000..bbc23f238e
--- /dev/null
+++ b/integrations/access/accessrequest/app_test.go
@@ -0,0 +1,98 @@
+package accessrequest
+
+import (
+	"context"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/wrappers"
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestOpsGenieGetMessageRecipients(t *testing.T) {
+	a := App{pluginType: types.PluginTypeOpsgenie, bot: testBot{}}
+	ctx := context.Background()
+	tests := []struct {
+		name               string
+		annotations        map[string][]string
+		expectedRecipients []common.Recipient
+	}{
+		{
+			name:               "no annotation",
+			annotations:        map[string][]string{},
+			expectedRecipients: []common.Recipient{},
+		},
+		{
+			name: "just notify-schedules",
+			annotations: map[string][]string{
+				types.TeleportNamespace + types.ReqAnnotationNotifySchedulesLabel: {"foo", "bar"},
+			},
+			expectedRecipients: []common.Recipient{
+				{
+					Name: "foo",
+					ID:   "foo",
+				},
+				{
+					Name: "bar",
+					ID:   "bar",
+				},
+			},
+		},
+		{
+			name: "just approval-schedules",
+			annotations: map[string][]string{
+				types.TeleportNamespace + types.ReqAnnotationApproveSchedulesLabel: {"foo", "bar"},
+			},
+			expectedRecipients: []common.Recipient{
+				{
+					Name: "foo",
+					ID:   "foo",
+				},
+				{
+					Name: "bar",
+					ID:   "bar",
+				},
+			},
+		},
+		{
+			name: "both notify and approval schedules",
+			annotations: map[string][]string{
+				types.TeleportNamespace + types.ReqAnnotationNotifySchedulesLabel:  {"foo", "bar"},
+				types.TeleportNamespace + types.ReqAnnotationApproveSchedulesLabel: {"baz", "hello"},
+			},
+			expectedRecipients: []common.Recipient{
+				{
+					Name: "foo",
+					ID:   "foo",
+				},
+				{
+					Name: "bar",
+					ID:   "bar",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := &types.AccessRequestV3{
+				Spec: types.AccessRequestSpecV3{
+					SystemAnnotations: wrappers.Traits(tt.annotations),
+				},
+			}
+			recipients := a.getMessageRecipients(ctx, req)
+			require.Equal(t, tt.expectedRecipients, recipients)
+		})
+	}
+
+}
+
+type testBot struct {
+	MessagingBot
+}
+
+func (testBot) FetchRecipient(ctx context.Context, recipient string) (*common.Recipient, error) {
+	return &common.Recipient{
+		Name: recipient,
+		ID:   recipient,
+	}, nil
+}
```
</Details>

I'm not sure if this bug is specific to the v15 backport or also on the master branch.

Reply:
Something like this look good to you?